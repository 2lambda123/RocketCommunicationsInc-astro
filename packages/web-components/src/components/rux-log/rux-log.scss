:host {
    display: block;
    overflow: hidden;
}

:host([hidden]) {
    display: none;
}

.rux-log__header-event-cell {
    width: 80%;
}

.header-event-container {
    display: flex;
    align-items: center;
}

.rux-log__filter {
    margin-left: auto;
}

rux-datetime {
    min-width: 4.875rem; //78px;
    display: inline-block;
}

rux-table-header-cell {
    width: 1%;
    border-width: 1px;
    border-style: solid;
    border-color: var(--color-background-base-default);
    height: calc(
        var(--spacing-10) + var(--spacing-1) - 2px
    ); //44 height minus border
}

rux-table-cell {
    height: calc(
        var(--spacing-8) - var(--spacing-050) - 1px
    ); //30 size - border
}

.rux-log__notification {
    width: 2000%;
    align-content: center;
    color: var(--color-text-inverse);
    background-color: #3a87cf;
    padding: 0.5rem;
}
