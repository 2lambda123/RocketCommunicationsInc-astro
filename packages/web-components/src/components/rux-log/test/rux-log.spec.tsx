import { newSpecPage } from '@stencil/core/testing'
import { RuxLog } from '../rux-log'
import { LogRow } from '../rux-log.model'

describe('rux-log', () => {
    it('renders', async () => {
        const page = await newSpecPage({
            components: [RuxLog],
            html: `<rux-log></rux-log>`,
        })

        const logData = [
            {
                timestamp: new Date(1557503698781),
                status: 'off',
                message: 'Antenna DGS 1 went offline.',
            },
            {
                timestamp: new Date(1557503698781),
                status: 'serious',
                message: 'USA-177 experienced solar panel misalignment.',
            },
            {
                timestamp: new Date(1557503698781),
                status: 'caution',
                message: 'USA-168 suffered power degradation.',
            },
        ] as LogRow[]

        page.root.data = logData
        await page.waitForChanges()
        expect(page.root).toMatchSnapshot()
    })

    it('filters', async () => {
        const logData = [
            {
                timestamp: new Date(1557503698781),
                status: 'off',
                message: 'Antenna DGS 1 went offline.',
            },
            {
                timestamp: new Date(1557503698781),
                status: 'serious',
                message: 'USA-177 experienced solar panel misalignment.',
            },
            {
                timestamp: new Date(1557503698781),
                status: 'caution',
                message: 'USA-168 suffered power degradation.',
            },
        ] as LogRow[]

        const log = new RuxLog()
        log.data = logData
        log.filter = 'Antenna DGS'

        expect(log.filteredData).toStrictEqual([
            {
                timestamp: new Date(1557503698781),
                status: 'off',
                message: 'Antenna DGS 1 went offline.',
            },
        ])
    })

    it('sets filter', async () => {
        const page = await newSpecPage({
            components: [RuxLog],
            html: `<rux-log></rux-log>`,
        })

        const input = page.root.shadowRoot.querySelector('rux-input')
        input.value = 'USA'
        await page.waitForChanges()
        input.dispatchEvent(new Event('ruxinput'))

        await page.waitForChanges()
        expect(page.root.filter).toBe('USA')
    })
})
