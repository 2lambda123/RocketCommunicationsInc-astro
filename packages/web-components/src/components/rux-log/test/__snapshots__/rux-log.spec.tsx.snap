// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-log renders 1`] = `
<rux-log>
  <mock:shadow-root>
    <slot name="table">
      <rux-table>
        <slot name="table-header">
          <rux-table-header class="relative">
            <slot name="table-header-row">
              <rux-table-header-row>
                <rux-table-header-cell>
                  Time
                </rux-table-header-cell>
                <rux-table-header-cell></rux-table-header-cell>
                <rux-table-header-cell class="rux-log__header-event-cell">
                  <div class="header-event-container">
                    Event
                    <rux-input class="rux-log__filter" placeholder="Search..." size="small" type="search"></rux-input>
                  </div>
                </rux-table-header-cell>
              </rux-table-header-row>
            </slot>
          </rux-table-header>
        </slot>
        <slot name="table-body">
          <rux-table-body>
            <rux-table-row>
              <rux-table-cell>
                <rux-datetime hour="2-digit" minute="2-digit" second="2-digit" time-zone="UTC"></rux-datetime>
              </rux-table-cell>
              <rux-table-cell>
                <rux-status status="off"></rux-status>
              </rux-table-cell>
              <rux-table-cell>
                Antenna DGS 1 went offline.
              </rux-table-cell>
            </rux-table-row>
            <rux-table-row>
              <rux-table-cell>
                <rux-datetime hour="2-digit" minute="2-digit" second="2-digit" time-zone="UTC"></rux-datetime>
              </rux-table-cell>
              <rux-table-cell>
                <rux-status status="serious"></rux-status>
              </rux-table-cell>
              <rux-table-cell>
                USA-177 experienced solar panel misalignment.
              </rux-table-cell>
            </rux-table-row>
            <rux-table-row>
              <rux-table-cell>
                <rux-datetime hour="2-digit" minute="2-digit" second="2-digit" time-zone="UTC"></rux-datetime>
              </rux-table-cell>
              <rux-table-cell>
                <rux-status status="caution"></rux-status>
              </rux-table-cell>
              <rux-table-cell>
                USA-168 suffered power degradation.
              </rux-table-cell>
            </rux-table-row>
          </rux-table-body>
        </slot>
      </rux-table>
    </slot>
  </mock:shadow-root>
</rux-log>
`;
